Collection interface

luizfgoncalves opened the pull request:
Implemented the basic Collection interface. @gcmgomes and @raphaelcampos can you guys review it?
Some observations:
- I didn't create a method  to get a subset of the collection yet, how will we handle it? 
- I left the function "*Point GetPoint()" commented because I can't return a *Point, how should I do it?
- Finally, on the "GetFeatureValues()" function, maybe we should use a std::set, as we are working with unique values.

luizfgoncalves committed:
Collection interface

luizfgoncalves committed:
Update collection.h

gcmgomes commented:
You only needed something like an accessor method `virtual const Point* GetPoint(unsigned point_id) const`. Anything else we can build as we need it. Maybe a global `GetLabel` method could be useful for the label set could be useful.

luizfgoncalves commented:
I followed the requirements @raphaelcampos included in the issue to do it. So, should I delete all the functions and create only these 2 you mentioned? (Then all the other methods will only be included in the concrete classes)

gcmgomes commented:
Worry about these 2 first, and then defining the others should be pretty straightforward.

luizfgoncalves commented:
I included the two functions and deleted the others. Wasnt sure if you mean GetLabel() should return the label given a point_id or just return all possible labels in a vector. Added both, will delete the wrong one in the next fix.
I was talking with @clebsonc, and we thought collection should handle the input file. (Including a function called GetInputData(), or something like that). What do you think?

luizfgoncalves committed:
Collection fixes

raphaelcampos commented:
About the size of the collection and the number of features?
Why does it not have a method to get a set of points given a set of ids? Similar to what you have done for labels?

luizfgoncalves commented:
I'm a little confused here. What we need to have on this interface? First I was following the issue list, and then @gcmgomes said we should just include only these 2 functions, and worry about other things later. What are we going to include on the interface, and what we are going to leave only to the concrete class implementation?

raphaelcampos commented:
I think it is necessary to have the method I mentioned before that will be implemented in the concrete class.
Dont you think we need to know the size and the number of features a dataset has?

luizfgoncalves commented:
I'm not used to work with interfaces so I'm not really sure what need to be here and what should just appear on the concrete implementation. I agree these two functions are important (I included them in the earlier version) and I feel they are generic enough, but maybe there's a reason to put them only on the concrete class.
